### PR TITLE
Release mechanism added for release on git tag v*

### PR DESCRIPTION
**Pull request by fahad-israr**
- setup workflow that on git tag v* (i.e. git tag v0.0.1) builds and create a draft release. 
- Issue: #22

### CODE REVIEW

File: .github/workflows/tag-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    strategy:
+        matrix:
+            os: [ubuntu-latest, windows-2019, macos-latest ]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get the version
+      id: get_version
+      run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+      shell: bash
+    - name: Create Release
```
Comment (maxandersen) [inline .github/workflows/tag-and-release.yml:29]: should not create the release before builds have completed imo.

File: .github/workflows/tag-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    - if: matrix.os == 'ubuntu-latest' 
+      name: Upload native executable Linux
+      id: upload-native-executable-linux
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ needs.createrelease.outputs.release_url }}
+        asset_path: ./cli/target/starfish-${{needs.createrelease.outputs.version}}.zip 
+        asset_name: starfish-linux.zip
```
Comment (maxandersen) [inline .github/workflows/tag-and-release.yml:99]: this shuold have the version number in it. why not upload the same as it it is named ?

Comment (fahad-israr) [reply .github/workflows/tag-and-release.yml]: Done that @Commit: https://github.com/maxandersen/starfish/pull/25/commits/873cb86d2802bc873430b8a16a1a43e3dd7052c5

File: .github/workflows/tag-and-release.yml (hunk 1 above)
Comment (maxandersen) [reply .github/workflows/tag-and-release.yml]: @fahad-israr you seem to still be creating the release before the build completed ?

Comment (maxandersen) [reply .github/workflows/tag-and-release.yml]: actually - I guess it is tricky doing this afterwards thus lets just live with it as its just a draft which we can delete if build fails for soem reason.

File: .github/workflows/tag-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+        shell: bash
+      - name: create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ steps.get_version.outputs.VERSION  }}
+          release_name: Release ${{ steps.get_version.outputs.VERSION  }}
```
Comment (maxandersen) [inline .github/workflows/tag-and-release.yml:35]:
```suggestion
          release_name: ${{ steps.get_version.outputs.VERSION  }}
```

no need to have "Release" in the Release.

**Pull request merged**